File: backend/.view/js/comment.js
import { socket } from "./websocket.js";
import { userStore } from "./user_store.js";

export const GrabCommentData = () => {
  const commentForm = document.querySelector(".comment_form");
  const text = commentForm.querySelector('textarea[name="text"]');

  if (text.value == "") {
    alert("Please write a comment");
    return;
  }

  const data = {
    type: "addcomment",
    userid: userStore.userid,
    postid: userStore.postid,
    text: text.value,
  };
  socket.send(JSON.stringify(data));
};

File: backend/internal/model/comment.go
package model

type Comment struct {
	ID           int    `json:"id"`
	UserId       int    `json:"user_id"`
	PostId       int    `json:"post_id"`
	Author       string `json:"author"`
	Text         string `json:"text"`
	Image         string `json:"image"`
	CreationDate string `json:"creation_date"`
}
